packages/flutter/test/gestures/gesture_binding_test.dart

```diff
@@ -397,4 +397,22 @@ void main() {
     expect(events[1], isA<PointerPanZoomUpdateEvent>());
     expect(events[2], isA<PointerPanZoomEndEvent>());
   });
+
+  test('Error handling', () {
+    const ui.PointerDataPacket packet = ui.PointerDataPacket(
+      data: <ui.PointerData>[
+        ui.PointerData(change: ui.PointerChange.down),
+        ui.PointerData(change: ui.PointerChange.up),
+      ],
+    );
+
+    final List<String> events = <String>[];
+    binding.callback = (PointerEvent event) { throw Exception('test: $event'); };
+    FlutterError.onError = (FlutterErrorDetails details) { events.add(details.toString()); };
+
+    GestureBinding.instance.platformDispatcher.onPointerDataPacket?.call(packet);
+    expect(events, hasLength(2));
```

Comment:
Could you assert that the error you get here is actually the one you expect (i.e. contains "while handling a pointer data packet")

(I just had to deal with a similar test that was expecting an error and caught an error, but it was an unrelated crash. So fixing the crash made the test fail because the error it originally wanted had long been removed...)

Reply:
excellent point, will do

Reply:
btw, your advice was excellent, turns out that the test WAS passing for the wrong reason, i was throwing in the wrong place and some unrelated code was catching the exception!

Reply:
Nice!